### ISSUE TITLE
refmt doesn't break all logical expressions in a long chain of them

### ISSUE THREAD

**Issue by kyldvs** [facebook/reason#1065]
```
let not_blacklisted str =>
  str != "dataAction" &&
  str != "dataContainer" &&
  str != "dataProvider" &&
  str != "dataReducer" &&
  str != "dataSelector";
```

becomes

```
let not_blacklisted str =>
  str != "dataAction" &&
  str != "dataContainer" &&
  str != "dataProvider" && str != "dataReducer" && str != "dataSelector";
```

**Comment by yunxing** [comment]
Looks like this is caused by we recursively group the items into labels: 
![image](https://cloud.githubusercontent.com/assets/1017689/23083765/234c2a7e-f514-11e6-9e9a-810cfc6c4031.png)

Ideally we should make those predicates into a flatted list, and use "&&" to separate them. 

cc @jordwalke who is more familiar with this part of logic. Do you have any quick workarounds for this problem?

**Comment by chenglou** [comment]
Fixed by #1259
